modules/security/pages/audit-privileged-actions.adoc

```diff
@@ -0,0 +1,144 @@
+= Auditing Privileged User Actions
+:description: Instructions on how to extract logs to audit user actions with the awk commandline utility.
+
+TigerGraph records every privileged user action that occurs on a TigerGraph deployment.
+This document walks you through the steps to identify privileged actions that occur on a deployment by user and by action.
+
+== Prerequisite
+* https://www.gnu.org/software/gawk/manual/gawk.html[The `awk` command-line utility] is installed on your server.
+** The commands on this page use `awk` to filter logs.
+If you are proficient in other text-processing utilities, you can use those instead, but the commands on this page won't apply.
+
+== Extract all privileged actions on an instance
+
+Run the following command in a bash shell to extract information on all privileged actions on an instance:
+
+[.wrap,console]
+----
+$ awk '/^(E|I|D)@.*/ {p=0}; /^(E|I|D)@.*_ACCESS_CHECK_.*/ {p=1; printf FILENAME ":" FNR ":"}; p { print }' \
+~/tigergraph/log/gsql/INFO.* <1>
+----
+<1> This pattern contains the default log location.
+If your logs are located at a different path, replace it with your log location.
+
+For example:
+
+[tabs]
+====
+Command::
++
+--
+[source.wrap,console]
+$ awk '/^(E|I|D)@.*/ {p=0}; /^(E|I|D)@.*_ACCESS_CHECK_.*/ {p=1; printf FILENAME ":" FNR ":"}; p { print }' \
+~/tigergraph/log/gsql/INFO.*
+--
+Results::
++
+--
+[source,text]
+/home/tigergraph/tigergraph/log/gsql/INFO.20220914-163940.595:1200:I@20220914 16:44:11.344 tigergraph|127.0.0.1:34994|00000000013 (PermissionUtil.java:371) User Name: tigergraph, _ACCESS_CHECK_RBAC_, Command Type: create query hello ( ) { print hello ; }, Validation Result: SUCCESS, Required RBAC Permission: [], Required Data Access Permission: [], Required Application Access Permission: []
+/home/tigergraph/tigergraph/log/gsql/INFO.20220914-163940.595:1532:I@20220914 19:27:34.585 tigergraph|127.0.0.1:41468|00000000013 (PermissionUtil.java:371) User Name: tigergraph, _ACCESS_CHECK_RBAC_, Command Type: /gsql/abortclientsession, Validation Result: SUCCESS, Required RBAC Permission: [], Required Data Access Permission: [], Required Application Access Permission: []
+
+--
+====
+
+== Extract logs by user
+
+Run the following command in a bash shell to extract information on all privileged actions on an instance by a particular user.
+Replace `<username>` in the first line with the name of the user whose actions you want to look up:
```

Comment:
Might worth mentioning that <username> is also case sensitive here in the example command.

`awk` does have some mechanism for allowing case insensitive regex pattern matching. But it may be beyond the scope of this page's example scope.